Utilities/ParallelSpawner.cs

```diff
@@ -48,13 +49,69 @@ public void Increment(IMyEntity entity)
             if (_spawned.Count < _maxCount)
                 return;
 
-            foreach (var g in _spawned)
+            foreach (MyCubeGrid g in _spawned)
             {
+                g.AddedToScene += EnablePower;
+                g.AddedToScene += EnableDampeners;
+                g.AddedToScene += EnableThrusters;
                 MyAPIGateway.Entities.AddEntity(g, true);
```

Comment:
Instead of setting methods to run on added to the scene and recasting it to a cube grid, im going to change to run on the callback. (Saves you the casting).

Ultimately this would work fine, just better to run your methods as CubeGrids on the spawned callback action. Will merge it for now, but ill change it before I update versions